app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -283,6 +287,26 @@ case class EvaluationResults(
     loop(typeInfo.expr)
   }
 
+  private def computePageLink(forPageId: PageId, evaluationContext: EvaluationContext) = {
+    val forModelPageId = forPageId.modelPageId
+    evaluationContext.pageIdSectionNumberMap.get(forModelPageId) match {
+      case Some(sectionNumber) =>
+        uk.gov.hmrc.gform.gform.routes.FormController
+          .formSection(evaluationContext.formTemplateId, evaluationContext.maybeAccessCode, sectionNumber)
+          .url
+      case None =>
+        evaluationContext.pageIdSectionNumberMap.toList
+          .find { case (modelPageId, _) =>
```

Comment:
Is this for add to list pages?
At the first sight this looks like this finds random page, am I missing something?

Reply:
Good point, added sorting by index